### DISCUSSION TITLE
Surfshark wireguard mtu value?

### DISCUSSION THREAD

**Discussion by Freethefire** [qdm12/gluetun#1875] [Q&A] [answered]
Hey, 

I'm going a bit nuts trying to figure out optimal mtu value for wireguard with surfshark. I've tried the old ping routine, but I'm only getting 100% package loss (ping -c 2 -s <mtu> <host>) on all but the lowest values ie 76 bits.
On stock mtu I'm getting decent speed at first but steadily drops down to at best around 1 Mbit (server capped at 9 Mbit). 
OpenVPN works well but eats too much CPU. 

Does anyone have any ideas on "correct" mtu or a script so that I can test more than on value at a time?

**Comment by frepke** [comment]
I haven't changed anything on the mtu settings. On my Surfshark/wireguard setup my download speed is 300+Mbit

**Reply by Freethefire** [reply]
Huh, strange. 
My connection works better when downloading over http for example. With torrents on the other hand the result is far from appealing. 
But when playing with the mtu value I'm getting both better and worse results, but I can't seem to find the sweet spot. 

Thanks for your reply!

**Comment by Freethefire** [comment, marked as answer]
Ok, seems to be working at last. 
Since Surfshark has dedicated P2P servers I used one of those before, there are 3 in neighboring countries. 
Instead I connected to the server in my country and set the MTU to 1420 (my network uses 1500 and then minus 80 for wireguard).
